src/Database.hs

```diff
@@ -13,29 +15,38 @@ import Database.PostgreSQL.Simple.SqlQQ
 import Database.PostgreSQL.Simple.Types
 import Types
 
-connectApp :: DatabaseConfig -> IO Connection
-connectApp (DBConfig host db user password dir) = connectPostgreSQL $ BS8.pack $ unpack url
+connectApp :: MonadIO m => DatabaseConfig -> m Connection
+connectApp (DBConfig host db user password dir) =
+  liftIO $ connectPostgreSQL $ BS8.pack $ unpack url
   where url = "host=" <> host <> " dbname=" <> db <> " user=" <> user <> " password=" <> password
 
-migrate :: FilePath -> Connection -> IO (MigrationResult String)
+migrate :: MonadIO m => FilePath -> Connection -> m (MigrationResult String)
 migrate dir conn = do
-  withTransaction conn $ runMigrations True conn
+  liftIO $ withTransaction conn $ runMigrations True conn
     [ MigrationInitialization
     , (MigrationDirectory dir)
     ]
- 
-validate :: FilePath -> Connection -> IO (MigrationResult String)
+
+validate :: MonadIO m => FilePath -> Connection -> m (MigrationResult String)
 validate dir conn = do
-  withTransaction conn $ runMigration $ MigrationContext
+  liftIO $ withTransaction conn $ runMigration $ MigrationContext
    (MigrationValidation (MigrationDirectory dir)) True conn
 
 instance FromRow ClientInfo where
   fromRow = ClientInfo <$> field <*> field <*> field <*> field <*> (fromPGArray <$> field)
 
-clients :: Connection -> IO [ClientInfo]
-clients conn = do
-  clients <- query_ conn clientsQueryString
-  return clients
+-- clients :: MonadIO m => ReaderT Connection m [ClientInfo]
```

Comment:
I think it would be nice to have a `runDb` function, which makes it super easy to run a db query.  You can see how this is being used in `Server.hs`.

If you use the `query_` function a lot, you might also want to make a corresponding wrapper function for it.